tempesta_fw/http_frame.c

```diff
@@ -1177,6 +1183,17 @@ tfw_h2_frame_type_process(TfwH2Ctx *ctx)
 	T_DBG3("%s: hdr->type=%hhu, ctx->state=%d\n", __func__, hdr->type,
 	       ctx->state);
 
+	if (unlikely(ctx->hdr.length > ctx->lsettings.max_frame_sz))
+		goto conn_term;
+
+	/*
+	 * TODO: RFC 7540 Section 6.2:
+	 * A HEADERS frame without the END_HEADERS flag set MUST be followed
+	 * by a CONTINUATION frame for the same stream. A receiver MUST treat
+	 * the receipt of any other type of frame or a frame on a different
+	 * stream as a connection error (Section 5.4.1) of type PROTOCOL_ERROR.
+	 */
```

Comment:
> The verification of HEADERS/CONTINUATION frame sequence is done in tfw_h2_stream_fsm()

Yes, no problem there. 

> The different streams intersection

Yes, that is what I meant. If peer has sent a `HEADER` frame without `HEADERS_END` flag, then he can't sent messages on other streams until the headers are closed with `CONTINUATION` frames followed by `HEADERS_END` flag.

>  e.g. via implementing the active_stream number in TfwH2Ctx structure, and setting/zeroing it for appropriate stream states in tfw_h2_stream_fsm().

Seems like it's enough to set some `ON_CONTINUATION` flag for the whole connection and check that the current stream ID is the same as the `lstream_id`

Reply:
The problem with `lstream_id` - is that it will not allow to handle this case for requests (i.e. streams) with HEADERS/CONTINUATION frames after DATA frames.